control-announcements/src/main/java/org/oskari/announcements/helpers/AnnouncementsDBHelper.java

```diff
@@ -0,0 +1,367 @@
+                }
+            }
+        }
+
+        JSONObject json = new JSONObject();
+        json.put("data", results);
+
+        return json;
+
+    }
+
+    /**
+     * Handle getting admin-announcements from database
+     * @param params
+     * @return
+     * @throws JSONException
+     * @throws ActionParamsException
+     */
+    public static JSONObject getAdminAnnouncements() throws JSONException, ActionParamsException {
+        Connection conn = null;
+
+        JSONArray results = new JSONArray();
+        String sql = "";
+        String sqlWithParams = "";
+
+        try  {
+            conn = getConnection();
+            StringBuilder sb = new StringBuilder();
+            sb.append("SELECT id, title, content, begin_date, end_date, active ");
+            sb.append("FROM oskari_announcements ");
+            sb.append("ORDER BY id DESC;");
+
+            sql = sb.toString();
+            PreparedStatement pstmt = conn.prepareStatement(sql);
+
+            sqlWithParams = pstmt.toString();
+            ResultSet rs = pstmt.executeQuery();
+            SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd");
+            while (rs.next()) {
+                JSONObject row = new JSONObject();
+                row.put("id", rs.getInt("id"));
+                row.put("title", rs.getString("title"));
+                row.put("content", rs.getString("content"));
+                row.put("begin_date", df.format(rs.getDate("begin_date")));
+                row.put("end_date", df.format(rs.getDate("end_date")));
+                row.put("active", rs.getBoolean("active"));
+                results.put(row);
+            }
+
+        } catch (SQLException e) {
+            LOG.error(e, "Cannot create SQL query, sql=" + sqlWithParams);
+
+        } finally {
+            if(conn != null) {
+                try {
+                    conn.close();
+                } catch (Exception e) {
+                    LOG.error(e);
+                }
+            }
+        }
+
+        JSONObject json = new JSONObject();
+        json.put("data", results);
+
+        return json;
+
+    }
+
+    /**
+     * Handle saving announcement to database
+     * @param params
+     * @return
+     * @throws JSONException
+     * @throws ActionParamsException
+     */
+    public static JSONObject saveAnnouncement(ActionParameters params) throws JSONException, ActionParamsException {
+        Connection conn = null;
+
+        JSONArray results = new JSONArray();
+        String sql = "";
+        String sqlWithParams = "";
+
+        try  {
+        
+            conn = getConnection();
+            StringBuilder sb = new StringBuilder();
+            sb.append("INSERT INTO oskari_announcements(title, content, begin_date, end_date, active) ");
+            sb.append("VALUES (? ::VARCHAR, ? ::VARCHAR, ? ::DATE, ? ::DATE, ? ::BOOLEAN) ");
+            sb.append("RETURNING id;");
+
+            
+
+            sql = sb.toString();
```

Comment:
I can't really see a reason why would you store the sql to this variable instead of calling `prepareStatement(sb.toString())`.